File: src/main/java/info/matsumana/tsujun/model/ResponseTable.java
package info.matsumana.tsujun.model;

import java.util.Arrays;

public class ResponseTable extends ResponseBase {

    private Object[] data;

    public Object[] getData() {
        return data;
    }

    public void setData(Object[] data) {
        this.data = data;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) { return true; }
        if (o == null || getClass() != o.getClass()) { return false; }
        if (!super.equals(o)) { return false; }
        ResponseTable that = (ResponseTable) o;
        return Arrays.equals(data, that.data);
    }

    @Override
    public int hashCode() {

        int result = super.hashCode();
        result = 31 * result + Arrays.hashCode(data);
        return result;
    }

    @Override
    public String toString() {
        return "ResponseTable{" +
               "data=" + Arrays.toString(data) +
               '}';
    }
}

File: src/main/kotlin/info/matsumana/tsujun/model/ResponseTable.kt
package info.matsumana.tsujun.model

import java.util.*

data class ResponseTable(val sequence: Int,
                         val sql: String,
                         val mode: Int,
                         val data: Array<Any>) {

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as ResponseTable

        if (sequence != other.sequence) return false
        if (sql != other.sql) return false
        if (mode != other.mode) return false
        if (!Arrays.equals(data, other.data)) return false

        return true
    }

    override fun hashCode(): Int {
        var result = sequence
        result = 31 * result + sql.hashCode()
        result = 31 * result + mode
        result = 31 * result + Arrays.hashCode(data)
        return result
    }
}
